CMakeLists.txt

```diff
@@ -0,0 +1,289 @@
+        "${PROJECT_SOURCE_DIR}/horovod/common/half.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/logging.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/message.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/operations.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/parameter_manager.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/response_cache.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/stall_inspector.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/thread_pool.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/timeline.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/tensor_queue.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/ops/collective_operations.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/ops/operation_manager.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/optim/bayesian_optimization.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/optim/gaussian_process.cc"
+        "${PROJECT_SOURCE_DIR}/horovod/common/utils/env_parser.cc")
+
+# Default Macro
+add_definitions(-DEIGEN_MPL2_ONLY=1)
+
+# Remove platform default std
+string(REGEX REPLACE  "-std=[^ ]+" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+# Pickup ar from environmental variable if available
+if(DEFINED ENV{AR})
+    set(CMAKE_AR $ENV{AR})
+endif()
+# Make CMAKE read CPPFLAGS
+set(CMAKE_CXX_FLAGS "$ENV{CPPFLAGS} ${CMAKE_CXX_FLAGS}")
+# Add default project CXX flags
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -fPIC -O3 -Wall -ftree-vectorize -g")
+# Add architecture specific optimization flags
+set(ARCH_FLAGS "-mf16c" "-mavx" "-mfma")
+set_build_arch_flags("${ARCH_FLAGS}")
+# Specify Horovod exports
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-exported_symbols_list,${CMAKE_SOURCE_DIR}/horovod.exp")
+else()
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--version-script=${CMAKE_SOURCE_DIR}/horovod.lds")
+endif()
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-Bsymbolic-functions -Wl,-z,relro")
+# GPU Operations
+set(HOROVOD_GPU $ENV{HOROVOD_GPU})
+set(HOROVOD_GPU_OPERATIONS $ENV{HOROVOD_GPU_OPERATIONS})
+if(DEFINED HOROVOD_GPU_OPERATIONS AND NOT "${HOROVOD_GPU_OPERATIONS}" MATCHES "^(MPI|NCCL)$")
+    message(FATAL_ERROR "HOROVOD_GPU_OPERATIONS=${HOROVOD_GPU_OPERATIONS} is invalid, supported values are '', 'MPI', and 'NCCL'.")
+endif()
+set_gpu_op(HOROVOD_GPU_ALLREDUCE "MPI;NCCL;DDL")
+set_gpu_op(HOROVOD_GPU_ALLGATHER "MPI;NCCL")
+set_gpu_op(HOROVOD_GPU_BROADCAST "MPI;NCCL")
+set_gpu_op(HOROVOD_GPU_ALLTOALL "MPI;NCCL")
+
+foreach(VAR in ITEMS HOROVOD_GPU_ALLREDUCE HOROVOD_GPU_ALLGATHER HOROVOD_GPU_BROADCAST HOROVOD_GPU_ALLTOALL)
+    if(DEFINED ${VAR})
+        string(SUBSTRING ${${VAR}} 0 1 ${VAR})
+        convert_to_ascii_dec(ASCII_DEC ${${VAR}})
+        add_definitions(-D${VAR}=${ASCII_DEC})
+    endif()
+endforeach()
+
+# PYTHON
+find_package(PythonInterp "3.6" REQUIRED)
```

Comment:
It seems `PythonInterp` is deprecated.  With this one, it was unable to resolve virtualenv paths.  Instead, using `find_package(Python3 ...)` fixed the issue in more recent versions of CMake.